### PR TITLE
feat(linux): comments on module u16 🐘

### PR DESCRIPTION
**Pull request by SabineSIL**
fixes parts of PR https://github.com/keymanapp/keyman/pull/9384.
Some comments may have been moved into an issue https://github.com/keymanapp/keyman/issues/11802

This PR addresses and bundles comments made on https://github.com/keymanapp/keyman/pull/9384 for the module u16

 [double function](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634481739)
 [misleading name](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634482671)
 [sz naming](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634483963)
 [save c_str() call](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634496807)
 [start with /0](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634506000)
 [wrong comment](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634510090)
 [naming](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634904457)
 [remove ToHex](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634917810)
 [remove comments](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634918498)

[unit tests](https://github.com/keymanapp/keyman/pull/9384#discussion_r1634512508)
@keymanapp-test-bot skip

### CODE REVIEW

File: linux/mcompile/keymap/u16.cpp [hunk 1]
```diff
@@ -25,6 +25,8 @@ std::vector<std::u16string> convert_argvW_to_Vector_u16str(int argc, wchar_t* ar
   return vector_u16;
 }
 
+// ToDo codecvt needs to be replaced !!
```
Comment (ermshiperete) [inline linux/mcompile/keymap/u16.cpp:28]: nit: the convention is to write tokens at the start of a comment like todo, review, fixme, refactor, note etc in all-caps.

```suggestion
// TODO: codecvt needs to be replaced !!
```

File: linux/mcompile/keymap/u16.h [hunk 2]
```diff
@@ -19,10 +19,10 @@ std::u16string u16string_from_string(std::string const str);
 std::u16string u16string_from_wstring(std::wstring const wstr);
 std::string string_from_u16string(std::u16string const str);
 
-std::wstring u16fmt(const KMX_WCHAR * str);
 void u16sprintf(KMX_WCHAR * dst, const size_t sz, const wchar_t* fmt, ...);
 
 std::wstring  convert_pchar16T_To_wstr(KMX_WCHAR *Name);
+std::wstring  convert_pchar16T_To_wstr(const KMX_WCHAR *Name);
```
Comment (ermshiperete) [inline linux/mcompile/keymap/u16.h:25]: I can only see one implementation of `convert_pchar16T_To_wstr`, so I think we should get rid of the definition without `const`:

```suggestion
std::wstring  convert_pchar16T_To_wstr(const KMX_WCHAR *Name);
```

**Review by ermshiperete — CHANGES_REQUESTED** [review]
Almost LGTM :smile:

File: linux/mcompile/keymap/u16.cpp (hunk 1 above)
Comment (SabineSIL) [reply linux/mcompile/keymap/u16.cpp]: OK-done

File: linux/mcompile/keymap/u16.h (hunk 2 above)
Comment (SabineSIL) [reply linux/mcompile/keymap/u16.h]: OK-done

**Review by ermshiperete — APPROVED** [review]

**Pull request merged**